### PR TITLE
Mask USB interrupt to avoid hardware bug in OpenTitan

### PR DESCRIPTION
**Pull request by phil-levis**
### Pull Request Overview

USB hardware on current OT master branch seems to have interrupt bugs: running Alarms causes persistent USB CONNECTED interrupts that can't be masked from USBDEV and cause the system to hang. So this PR modifies the PLIC to enable all interrupts *except* for the USB ones. Some open PRs on OT fix this, we'll re-enable USB interrupts when they are merged.
 
To observe this behavior, take a look at the  `opentitan-alarm-causes-usb-failures` branch.

This bug report came from the alarm redesign branch; separating USB interrupt handling into a separate PR seems advisable.

I've entered this as an issue for OpenTitan: https://github.com/lowRISC/opentitan/issues/3388

### Testing Strategy

Compiling and running the multi alarm test. 

### TODO or Help Wanted

None

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

**Comment by alistair23** [comment]
Can we link this with an OT issue? Otherwise I'm worried we'll loose track of it.

**Comment by phil-levis** [comment]
Working on it.

**Comment by phil-levis** [comment]
Done.

**Comment by alistair23** [comment]
Cool. Can you link to the issue in the comment?

**Review by bradjc — DISMISSED** [review]

File: chips/earlgrey/src/plic.rs [hunk 1]
```diff
@@ -41,9 +41,16 @@ pub unsafe fn clear_all_pending() {
...
-    for enable in plic.enable.iter() {
-        enable.set(0xFFFF_FFFF);
-    }
+
+    // USB hardware on current OT master branch seems to have
+    // interrupt bugs: running Alarms causes persistent USB
+    // CONNECTED interrupts that can't be masked from USBDEV and
+    // cause the system to hang. So enable all interrupts except
+    // for the USB ones. Some open PRs on OT fix this, we'll re-enable
+    // USB interrurupts.
```
Comment (bradjc) [inline chips/earlgrey/src/plic.rs:50]:
```suggestion
    // USB interrurupts.
    //
    // https://github.com/lowRISC/opentitan/issues/3388
```

**Review by alistair23 — APPROVED** [review]

**Pull request merged**